app/Models/ProcessTransaction.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+
+declare(strict_types=1);
+
+namespace App\Models;
+
+class ProcessTransaction
```

Comment:
Maybe a bit misleading name of class. ProcessTransaction can indicate processing actual transactions but this class simply does some calculations. Maybe call the class something like `CalculateTransactionTotals` or something like that